x/auth/keeper/grpc_query.go

```diff
@@ -83,3 +83,28 @@ func (ak AccountKeeper) Params(c context.Context, req *types.QueryParamsRequest)
 
 	return &types.QueryParamsResponse{Params: params}, nil
 }
+
+// ModuleAccounts returns all the existing Module Accounts
+func (ak AccountKeeper) ModuleAccounts(c context.Context, req *types.QueryModuleAccountsRequest) (*types.QueryModuleAccountsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	modAccounts := make([]*codectypes.Any, 0, len(ak.permAddrs))
+
+	for moduleName := range ak.permAddrs {
+		account := ak.GetModuleAccount(ctx, moduleName)
+		if account == nil {
+			return nil, status.Errorf(codes.NotFound, "account %s not found", moduleName)
+		}
+		any, err := codectypes.NewAnyWithValue(account)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, err.Error())
+		}
+		modAccounts = append(modAccounts, any)
```

Comment:
Since this is pre-allocated, I would use direct indexing instead of appending.

Reply:
👍

Reply:
Okiee :p